pkgs/development/tools/haskell/haskell-language-server/hls-ghcide.nix

```diff
@@ -5,21 +5,20 @@
 , ghc-boot-th, ghc-check, ghc-paths, ghc-typelits-knownnat, gitrev
 , haddock-library, hashable, haskell-lsp, haskell-lsp-types
 , hie-bios, hslogger, lens, lsp-test, mtl, network-uri
-, opentelemetry, optparse-applicative, prettyprinter
-, prettyprinter-ansi-terminal, process, QuickCheck
-, quickcheck-instances, regex-tdfa, rope-utf16-splay, safe
-, safe-exceptions, shake, sorted-list, stdenv, stm, syb, tasty
-, tasty-expected-failure, tasty-hunit, tasty-quickcheck
-, tasty-rerun, text, time, transformers, unix, unordered-containers
-, utf8-string, yaml
+, optparse-applicative, prettyprinter, prettyprinter-ansi-terminal
+, process, QuickCheck, quickcheck-instances, regex-tdfa
+, rope-utf16-splay, safe, safe-exceptions, shake, sorted-list
+, stdenv, stm, syb, tasty, tasty-expected-failure, tasty-hunit
+, tasty-quickcheck, tasty-rerun, text, time, transformers, unix
+, unordered-containers, utf8-string, yaml
 }:
 mkDerivation {
   pname = "ghcide";
-  version = "0.2.0";
+  version = "0.3.0";
```

Comment:
Not sure if that's an issue in itself, but having a single update.sh makes fixing or extending with  other libs (like this PR) into one non-atomic commit. While nixpkgs is a bit blurry about its "atomic changes" policy, I think it is still a good idea to split package update and build fixes.

@cdepillabout do you consider it to not be an issue?
Should this PR be split into more atomic commits manually? (my preference)
Do we change the workflow by splitting the update scripts?
Anything else?

Reply:
I split out the commits and added a note to update.sh to remind others to do the same.

Reply:
I don't particularly worry about it that much.  I agree that it is a good idea to have atomic changes, but there is some trade-off with making it unreasonably challenging for people to contribute.

I would be fine with the HLS stuff all going into one commit, since it is all basically related anyway.